src/burnfee.rs

```diff
@@ -0,0 +1,106 @@
+pub const HEARTBEAT: u64 = 30_000;
+/// The Burnfee object which contains our starting value and current value when the fee was paid
+#[derive(PartialEq, Clone, Debug)]
+pub struct BurnFee {
+    pub start: f64,
+    pub current: u64,
+}
+
+impl BurnFee {
+    /// Returns the BurnFee used to calculate the work needed to produce a block
+    ///
+    /// * `start` - y-value at x = 0
+    /// * `current` - y-value at x = 0 for next bloc
+    pub fn new(start: f64, current: u64) -> Self {
+        return Self {
+	      start,
+	      current
+	    };
+    }
+
+    /// Returns the amount of work needed to produce a block given the timestamp of
+    /// the previous block, the current timestamp, and the y-axis of the burn fee
+    /// curve. This is used both in the creation of blocks (mempool) as well as
+    /// during block validation.
+    ///
+    /// * `prevts` - timestamp of previous block
+    /// * `ts`     - candidate timestamp
+    /// * `start`  - burn fee value (y-axis) for curve determination ("start")
+    pub fn return_work_needed(prevts: u64, ts: u64, start: f64) -> u64 {
+	    let mut elapsed_time = ts - prevts;
+        if elapsed_time == 0 { elapsed_time = 1; }
+        if elapsed_time >= (2 * HEARTBEAT) { return 0; }
+
+        let elapsed_time_float     = elapsed_time as f64;
```

Comment:
Is this usage of `as` common in rust?  
Asking because in TS this would be a hint that typing might have issues, because your manually telling it it's `f64`.

Reply:
yes it is common and I don't see any problem here.

Reply:
There's a bigger issue here which is that I think this should be an int, not a float....